src/utils/validationFunctions.js

```diff
@@ -18,6 +18,30 @@ export function register(email, password) {
     throw Error(ERR.PASSWORD_REQUIREMENTS)
   }
 }
+const parseAndValidateDate = (dateString) => {
+  const [year, month, day] = dateString.split('/').map(Number)
+  const date = new Date(Date.UTC(year, month - 1, day))
+  if (!date.toISOString() && !date.startsWith(dateString.replace(/\//g, '-'))) {
+    return null
+  } else {
+    return date
+  }
+}
+export function dateValidation(startDate, endDate) {
+  if (!startDate || !endDate) {
+    throw Error(ERR.DATE_REQUIRED)
+  }
+  if (!dateRegex.test(startDate) || !dateRegex.test(endDate)) {
+    throw Error(ERR.DATE_FORMATTING)
+  }
+
+  const parsedStartDate = parseAndValidateDate(startDate)
+  const parsedEndDate = parseAndValidateDate(endDate)
+  if (!parsedStartDate || !parsedEndDate) {
+    throw Error(ERR.DATE_FORMATTING)
+  }
+  return { parsedStartDate, parsedEndDate }
+}
```

Comment:
There should be tests for things like this. You don't have to include them with this PR, but when this gets merged please add including tests to your list of work

Reply:
I'm afraid I do not understand these specs...

Reply:
Do you want me to put them in an .env?

Reply:
I'm just asking that you create unit tests for these functions in a future PR